python/cugraph/cugraph/experimental/datasets/dataset.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cugraph
+import cudf
+import yaml
+import os
+from pathlib import Path
+
+
+class DefaultDownloadDir:
+    def __init__(self):
+        self._dd = Path(os.environ.get("RAPIDS_DATASET_ROOT_DIR",
+                                       Path.home() / ".cugraph/datasets"))
+
+    @property
+    def dd(self):
+        if self._dd is None:
+            self._dd = Path(os.environ.get("RAPIDS_DATASET_ROOT_DIR",
+                                           Path.home() / ".cugraph/datasets"))
+        return self._dd
+
+    @dd.setter
+    def dd(self, new):
+        self._dd = new
+
+    def clear(self):
+        self._dd = None
+
+
+default_download_dir = DefaultDownloadDir()
+
+
+class Dataset:
+    """
+    A Dataset Object, used to easily import edgelist data and cuGraph.Graph
+    instances.
+
+    Parameters
+    ----------
+    meta_data_file_name : yaml file
+        The metadata file for the specific graph dataset, which includes
+        information on the name, type, url link, data loading format, graph
+        properties
+
+    """
+    def __init__(self, meta_data_file_name):
+        self.__dir_path = Path(__file__).parent.absolute()
+
+        self._meta_data_file_name = meta_data_file_name
+        self.__read_meta_data_file(self._meta_data_file_name)
+
+        self._dd = default_download_dir.dd
```

Comment:
This could probably also use a better name than `dd`, plus it would be better if you assigned it the `default_download_dir` instance itself instead of the value of `.dd`, since that will allow you to always easily lookup the current value instead of having a `__refresh_dd()` method.
```suggestion
        self._download_dir = default_download_dir
```